django_auth_adfs/backend.py

```diff
@@ -420,6 +420,11 @@ def authenticate(self, request=None, authorization_code=None, **kwargs):
 
         adfs_response = self.exchange_auth_code(authorization_code, request)
         access_token = adfs_response["access_token"]
+
+        # Extract claims before user lookup
+        claims = self.validate_access_token(access_token)
```

Comment:
Is there a reason you're not overriding `validate_access_token` in your code to achieve the logic you want?